compiler/noirc_frontend/src/lexer/token.rs

```diff
@@ -320,7 +321,7 @@ pub fn token_to_borrowed_token(token: &Token) -> BorrowedToken<'_> {
 #[derive(Clone, PartialEq, Eq, Hash, Debug, PartialOrd, Ord)]
 pub enum FmtStrFragment {
     String(String),
-    Interpolation(String, Span),
+    Interpolation(String, Span, FileId),
```

Comment:
Why not directly store a `Location` instead of separate `Span` and `FileId`?

Reply:
I forgot why so I tried it again and remembered: `Location` isn't `PartialOrd`/`Ord` and I wasn't sure if this was intentional or not. If there's no issue with that I can make the change.

Reply:
I think it was just never added because it wasn't needed before. Should be fine to add, although I'll add that it'd make more sense when comparing locations that the FileId is compared before the Span.

Reply:
Done!